src/lib/comms/sol-network-impl-contiki.c

```diff
@@ -161,20 +161,20 @@ sol_network_shutdown(void)
     return;
 }
 
-SOL_API bool
+SOL_API int
 sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_link *link,
     enum sol_network_event event),
     const void *data)
 {
-    return false;
+    return -errno;
```

Comment:
This is not quite correctly. You must return the errno value not the value of the errno variable.
What I mean, is that you need to do something like: `return -ENOSYS;`